src/IonEventStream.ts

```diff
@@ -301,6 +299,22 @@ export class IonEventStream {
             currentEvent['isNull'],
             currentEvent['value_text']
         );
+
+        let binaryEvent = this.eventFactory.makeEvent(
```

Comment:
This can go inside the `if(eventType === IonEventType.SCALAR){` block below since it's only needed in that case. To simplify it, you could also just do a data model equivalence comparison on `value_text` and `value_binary` if it's a scalar event. That way you don't need to create a dummy event for binary.